service/service_test.go

```diff
@@ -414,6 +414,29 @@ func TestServiceTelemetryLogger(t *testing.T) {
 	assert.NotNil(t, srv.telemetrySettings.Logger)
 }
 
+func TestServiceFatalError(t *testing.T) {
+	set := newNopSettings()
+	set.AsyncErrorChannel = make(chan error)
+
+	srv, err := New(context.Background(), set, newNopConfig())
+	require.NoError(t, err)
+
+	assert.NoError(t, srv.Start(context.Background()))
+	t.Cleanup(func() {
+		assert.NoError(t, srv.Shutdown(context.Background()))
+	})
+
+	go func() {
+		ev := component.NewFatalErrorEvent(assert.AnError)
+		srv.host.notifyComponentStatusChange(&component.InstanceID{}, ev)
+	}()
+
+	err = <-srv.host.asyncErrorChannel
+
+	require.Error(t, err)
+	require.ErrorIs(t, err, assert.AnError)
```

Comment:
is this needed if you're already requiring an error above it?

Reply:
Not needed. There were a few more instances of this that I also cleaned up.

Reply:
Thx